### PR TITLE
Pal 764 user service j unit5

### PR DESCRIPTION
**Pull request by developer01189998819991197253**

### CODE REVIEW

File: user-service/src/component-tests/java/uk/gov/gchq/palisade/component/user/service/CaffeineUserCachingTest.java [hunk 1]
```diff
@@ -61,45 +61,58 @@ private void forceCleanUp() {
...
-        // When
-        User addedUser = userService.addUser(user);
-        // Then
-        assertThat(addedUser).isEqualTo(user);
-
-        // When
-        User getUser = userService.getUser(user.getUserId().getId());
-        // Then
-        assertThat(getUser).isEqualTo(user);
+        // Then the user is added to the cache
```
Comment (pd104923) [inline user-service/src/component-tests/java/uk/gov/gchq/palisade/component/user/service/CaffeineUserCachingTest.java]:
```suggestion
        // When the user is added to the cache
```

File: user-service/src/component-tests/java/uk/gov/gchq/palisade/component/user/service/CaffeineUserCachingTest.java [hunk 2]
```diff
@@ -61,45 +61,58 @@ private void forceCleanUp() {
...
+        userService.addUser(user);
+        // When we retrieve the user from the cache
```
Comment (pd104923) [inline user-service/src/component-tests/java/uk/gov/gchq/palisade/component/user/service/CaffeineUserCachingTest.java]:
```suggestion
        // Then retrieve the user from the cache
```

File: user-service/src/component-tests/java/uk/gov/gchq/palisade/component/user/service/CaffeineUserCachingTest.java [hunk 3]
```diff
@@ -61,45 +61,58 @@ private void forceCleanUp() {
...
         // Given the user is not added to the cache
 
         // When
-        assertThrows(NoSuchUserIdException.class,
-                () -> userService.getUser("definitely-not-a-real-user"), "testNonExistentUser should throw noSuchIdException"
-        );
+        var noSuchUserIdException = assertThrows(NoSuchUserIdException.class,
+                () -> userService.getUser("definitely-not-a-real-user"), "testNonExistentUser should throw noSuchIdException");
+
+        assertThat(noSuchUserIdException)
```
Comment (pd104923) [inline user-service/src/component-tests/java/uk/gov/gchq/palisade/component/user/service/CaffeineUserCachingTest.java:87]:
```suggestion
        // Then check the returned exception
        assertThat(noSuchUserIdException)
```

File: user-service/src/component-tests/java/uk/gov/gchq/palisade/component/user/service/CaffeineUserCachingTest.java [hunk 4]
```diff
@@ -115,9 +128,14 @@ void testMaxSize() {
         forceCleanUp();
 
         // Then a NoSuchUserIdException is thrown as the User no longer exists
-        assertThrows(NoSuchUserIdException.class,
-                () -> userService.getUser(makeUserId.apply(0)), "testMaxSizeTest should throw noSuchIdException"
-        );
+        var noSuchUserIdException = assertThrows(NoSuchUserIdException.class,
+                () -> userService.getUser(makeUserId.apply(0)), "testMaxSizeTest should throw noSuchIdException");
+
+        assertThat(noSuchUserIdException)
```
Comment (pd104923) [inline user-service/src/component-tests/java/uk/gov/gchq/palisade/component/user/service/CaffeineUserCachingTest.java]:
```suggestion
        // When we get the user from the cache
        var noSuchUserIdException = assertThrows(NoSuchUserIdException.class,
                () -> userService.getUser(makeUserId.apply(0)), "testMaxSizeTest should throw noSuchIdException");

        // Then check the returned exception
        assertThat(noSuchUserIdException)
```

File: user-service/src/unit-tests/java/uk/gov/gchq/palisade/service/user/service/NullUserServiceTest.java [hunk 5]
```diff
@@ -31,21 +33,32 @@
...
+        // Given a new User is created
+        User user = new User().userId("testUser")
+                .addRoles(Collections.singleton("testRole"))
+                .addAuths(Collections.singleton("testAuth"));
+
+        // When we add the user to the Service, and the added user is returned
         User actual = nullUserService.addUser(user);
-        assertThat(user).isEqualTo(actual);
+
+        assertThat(user)
```
Comment (pd104923) [inline user-service/src/unit-tests/java/uk/gov/gchq/palisade/service/user/service/NullUserServiceTest.java:60]:
```suggestion
        // Then check the returned user has not been modified
        assertThat(user)
```

File: user-service/src/contract-tests/java/uk/gov/gchq/palisade/contract/user/rest/HealthActuatorContractTest.java [hunk 6]
```diff
@@ -57,7 +58,14 @@ void testServiceIsHealthy() {
         // When we GET the /actuator/health REST endpoint (used by k8s)
         final ResponseEntity<String> health = restTemplate.getForEntity("/actuator/health", String.class);
 
-        // Then the service reports itself to be healthy
-        assertThat(health.getStatusCode()).isEqualTo(HttpStatus.OK);
+        assertAll("Assert the Health Actuator",
```
Comment (pd104923) [inline user-service/src/contract-tests/java/uk/gov/gchq/palisade/contract/user/rest/HealthActuatorContractTest.java:62]:
```suggestion
        // Then check the service reports itself to be healthy
        assertAll("Assert the Health Actuator",
```

**Review by pd104923 — CHANGES_REQUESTED** [review]
Some small questions/changes

**Review by pd104923 — DISMISSED** [review]

File: user-service/src/component-tests/java/uk/gov/gchq/palisade/component/user/service/UserServiceAsyncProxyErrorTest.java [hunk 7]
```diff
@@ -55,64 +56,61 @@
```
Comment (dev930018) [inline user-service/src/component-tests/java/uk/gov/gchq/palisade/component/user/service/UserServiceAsyncProxyErrorTest.java:56]: Can't comment on the correct line (L51), but is this not just a `UserServiceAsyncProxyTest` as it tests successes and errors?

File: user-service/src/unit-tests/java/uk/gov/gchq/palisade/service/user/web/UserRestControllerTest.java [hunk 8]
```diff
@@ -87,10 +92,15 @@ void testAddAndGetUser() {
...
+        assertThat(user)
+                .as("Check that the User is the one we expect and has not been modified")
+                .usingRecursiveComparison()
+                .isEqualTo(expected);
 
         List<String> debugMessages = getMessages(event -> event.getLevel() == Level.INFO);
-        assertThat(debugMessages).isNotEmpty();
+        assertThat(debugMessages)
+                .as("Check there are logging messages at INFO level")
+                .isNotEmpty();
```
Comment (dev930018) [inline user-service/src/unit-tests/java/uk/gov/gchq/palisade/service/user/web/UserRestControllerTest.java]: Sounds a lot like `infoMessages` to me

**Review by dev930018 — CHANGES_REQUESTED** [review]

File: user-service/src/component-tests/java/uk/gov/gchq/palisade/component/user/service/UserServiceAsyncProxyErrorTest.java (hunk 7 above)
Comment (developer01189998819991197253) [reply user-service/src/component-tests/java/uk/gov/gchq/palisade/component/user/service/UserServiceAsyncProxyErrorTest.java:56]: updated

File: user-service/src/unit-tests/java/uk/gov/gchq/palisade/service/user/web/UserRestControllerTest.java (hunk 8 above)
Comment (developer01189998819991197253) [reply user-service/src/unit-tests/java/uk/gov/gchq/palisade/service/user/web/UserRestControllerTest.java]: you're correct! updated

**Review by dev930018 — DISMISSED** [review]

**Review by dev-958 — APPROVED** [review]

**Pull request merged**